core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -2732,6 +2732,7 @@ private CompiledQuery vacuum(SqlExecutionContext executionContext) throws SqlExc
             int tableNamePos = lexer.lastTokenPosition();
             CharSequence eol = SqlUtil.fetchNext(lexer);
             if (eol == null || Chars.equals(eol, ';')) {
+                executionContext.getCairoSecurityContext().checkWritePermission();
```

Comment:
I hope this is the right place to inject the check. alternatively, the check could be at the very to of the method, but I believe we should validate the syntax before doing any kind of security checks. anyway, I don't insist on this, some methods are already checking permission as the first thing. (and other do it later as they rely on `getWriter()`)

Reply:
Looks good to me .